### PR TITLE
GOVUKAPP-591 Forced Update

### PR DESCRIPTION
**Pull request by TimGJD**
Add app forced update screen and logic to show it

[Link to Jira](https://govukverify.atlassian.net/browse/GOVUKAPP-591)

[Link to Figma](https://www.figma.com/design/5u5cvVCgKY3Rng2ADNV30K/2024-07---GOV.UK-app---beta-designs?node-id=363-6730&node-type=section&t=ISmPGTBwHk6ZtgdH-0)

test_loadInNavigationController_dark_rendersCorrectly@3x

test_loadInNavigationController_light_rendersCorrectly@3x

### CODE REVIEW

File: Tests/govuk_ios/govuk_ios_unit_tests/Specs/Extensions/Foundation/String+ExtensionsTests.swift [hunk 1]
```diff
@@ -129,4 +129,23 @@ struct String_ExtensionsTests {
         #expect(expected == "Child benefit CD 98 76 54 A allowance".redactPii())
         #expect(expected == "Child benefit CD  98  76  54  A allowance".redactPii())
     }
+
+    @Test
+    func isVersion_returnsExpectedResult() {
```
[the remark this answers is not in the document]
Comment (batesyuk) [reply Tests/govuk_ios/govuk_ios_unit_tests/Specs/Extensions/Foundation/String+ExtensionsTests.swift:134]: Infact `TopicTests` might be a good one with the zip implementation for arguments

**Review by batesyuk — APPROVED** [review]

**Review by batesyuk — CHANGES_REQUESTED** [review]
Approved before by accident

File: Production/govuk_ios/Extensions/Foundation/Bundle+AppVersionProvider.swift
[the remark this answers is not in the document]
Comment (TimGJD) [reply Production/govuk_ios/Extensions/Foundation/Bundle+AppVersionProvider.swift:1]: Done

File: Production/govuk_ios/ViewModels/AppForcedUpdateContainerViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import UIComponents
+import UIKit
+
+class AppForcedUpdateContainerViewModel: ObservableObject {
+    private let urlOpener: URLOpener
+
+    let title = String.appAvailability.localized("forcedUpdateTitle")
+    let subheading = String.appAvailability.localized("forcedUpdateSubheading")
+    let updateButtonTitle = String.appAvailability.localized("updateButtonTitle")
+    let appStoreAppUrl = Constants.API.appStoreAppUrl
```
[the remark this answers is not in the document]
Comment (TimGJD) [reply Production/govuk_ios/ViewModels/AppForcedUpdateContainerViewModel.swift]: Done

File: Tests/govuk_ios/govuk_ios_unit_tests/Specs/Extensions/Foundation/String+ExtensionsTests.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (TimGJD) [reply Tests/govuk_ios/govuk_ios_unit_tests/Specs/Extensions/Foundation/String+ExtensionsTests.swift]: Done

File: Tests/govuk_ios/govuk_ios_unit_tests/Specs/Coordinators/AppForcedUpdateCoordinatorTests.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import UIKit
+import Testing
+
+@testable import govuk_ios
+
+@Suite
+@MainActor
+struct AppForcedUpdateCoordinatorTests {
+    @Test
+    func start_isAppForcedUpdate_falseCallsDismiss() async {
```
[the remark this answers is not in the document]
Comment (TimGJD) [reply Tests/govuk_ios/govuk_ios_unit_tests/Specs/Coordinators/AppForcedUpdateCoordinatorTests.swift]: Done

**Review by batesyuk — APPROVED** [review]

**Pull request merged**